File: src/main/java/com/gangofconnectfour/powerfourservice/repository/UserRepository.java
package com.gangofconnectfour.powerfourservice.repository;

import com.gangofconnectfour.powerfourservice.model.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {

    @Query("from User u where u.email = :mail")
    User getUserByMail(String mail);

    User findByUuid(Long uuid);
}

File: src/main/kotlin/com/gangofconnectfour/powerfourservice/repository/UserRepository.kt
package com.gangofconnectfour.powerfourservice.repository

import com.gangofconnectfour.powerfourservice.model.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query

interface UserRepository : JpaRepository<User, Long> {

    @Query("from User u where u.email = :mail")
    fun getUserByMail(mail : String): User

    fun findByUuid(uuid : Long) : User
}